### ISSUE TITLE
Support for database-level foreign keys

### ISSUE THREAD

**Issue by dankohn** [sequelize/sequelize#389]
It would be great for Sequelize to support real foreign keys. When ActiveRecord was created SQLite and MySQL didn't yet support foreign key constraints, so it's understandable why DHH didn't insist on them. But all 3 Sequelize databases now support database foreign keys, and they are a useful piece of safety equipment, especially if you ever run node cluster or access the database outside of Sequelize. Having those database constraints could preserve consistency. See the seatbelt analogy here: http://stackoverflow.com/questions/2589509/does-rails-need-database-level-constraints

Basically, I'm suggesting that it would be great if someone could port the Rails gem foreigner to Sequelize:
https://github.com/matthuhiggins/foreigner

And, adding immigrant support for migrations would be great too:
https://github.com/jenseng/immigrant
